hydra_agent/agent.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from hydra_agent.redis_proxy import RedisProxy
+from hydra_agent.graphutils_operations import GraphOperations
+from hydra_agent.hydra_graph import InitialGraph
+from hydra_python_core import doc_maker
+from requests import Session
+
+logger = logging.getLogger(__file__)
+
+
+class HydraAgent(Session):
+    def __init__(self, entrypoint_url):
+        self.entrypoint_url = entrypoint_url.strip().rstrip('/')
+        self.redis_proxy = RedisProxy()
+        self.redis_connection = self.redis_proxy.get_connection()
+        self.graph_operations = GraphOperations(entrypoint_url,
+                                                self.redis_proxy)
+        super().__init__()
+        jsonld_api_doc = super().get(self.entrypoint_url + '/vocab').json()
+        self.api_doc = doc_maker.create_doc(jsonld_api_doc)
+        self.initialize_graph()
+
+    def initialize_graph(self):
+        self.graph = InitialGraph()
+        self.redis_connection.delete("apigraph")
+        self.graph.main(self.entrypoint_url, self.api_doc, True)
+        self.redis_connection.sadd("fs:url", self.entrypoint_url)
+
+    def get(self, url):
+        response = self.graph_operations.get_resource(url)
```

Comment:
This is the first implementation to start using Redis, this gets already cached GET requests and return them. This avoids also duplications of getting the same resources into Redis more than once. The same function in the near future should receive somehow a property and also query by it.